File: DesignPatternsInAutomatedTesting-Series/PatternsInAutomation.Tests/Advanced/Strategy/Advanced/Strategies/VatTaxOrderPurchaseStrategy.cs
using System;
using PatternsInAutomatedTests.Advanced.Strategy.Advanced.Base;
using PatternsInAutomatedTests.Advanced.Strategy.Data;
using PatternsInAutomatedTests.Advanced.Strategy.Enums;
using PatternsInAutomatedTests.Advanced.Strategy.Pages.PlaceOrderPage;

namespace PatternsInAutomatedTests.Advanced.Strategy.Advanced.Strategies
{
    public class VatTaxOrderPurchaseStrategy : IOrderPurchaseStrategy
    {
        public VatTaxOrderPurchaseStrategy()
        {
            this.VatTaxCalculationService = new VatTaxCalculationService();
        }

        public VatTaxCalculationService VatTaxCalculationService { get; set; }

        public void ValidateOrderSummary(string itemsPrice, ClientPurchaseInfo clientPurchaseInfo)
        {
            Countries currentCountry = (Countries)Enum.Parse(typeof(Countries), clientPurchaseInfo.BillingInfo.Country);
            decimal currentItemPrice = decimal.Parse(itemsPrice);
            decimal vatTax = this.VatTaxCalculationService.Calculate(currentItemPrice, currentCountry);

            PlaceOrderPage.Instance.Validate().EstimatedTaxPrice(vatTax.ToString());
        }
        public void ValidateClientPurchaseInfo(ClientPurchaseInfo clientPurchaseInfo)
        {
           // Throw a new Argument exection if the country is not part of the EU Union.
        }
    }
}

File: DesignPatternsInAutomatedTestingJava-Series/src/test/java/strategyadvanced/strategies/VatTaxOrderPurchaseStrategy.java
package strategyadvanced.strategies;

import strategyadvanced.base.OrderPurchaseStrategy;
import strategyadvanced.core.Driver;
import strategyadvanced.data.PurchaseInfo;
import strategyadvanced.enums.Country;
import strategyadvanced.pages.checkoutpage.CheckoutPage;
import strategyadvanced.services.VatTaxCalculationService;

import java.util.Arrays;

public class VatTaxOrderPurchaseStrategy implements OrderPurchaseStrategy {
    private final VatTaxCalculationService vatTaxCalculationService;

    public VatTaxOrderPurchaseStrategy() {
        vatTaxCalculationService = new VatTaxCalculationService();
    }

    @Override
    public void assertOrderSummary(double itemPrice, PurchaseInfo purchaseInfo) {
        var currentCountry = Arrays.stream(Country.values())
                .filter(country -> country.toString().equals(purchaseInfo.getCountry()))
                .toArray(Country[]::new)[0];
        var vatTax = vatTaxCalculationService.calculate(itemPrice, currentCountry, purchaseInfo);

        var checkoutPage = new CheckoutPage();
        Driver.waitForAjax();
        Driver.waitUntilPageLoadsCompletely();
        checkoutPage.assertions().assertOrderVatTaxPrice(vatTax);
    }

    @Override
    public void validatePurchaseInfo(PurchaseInfo purchaseInfo) {
        // Throw a new IllegalArgumentException if the country is not part of the EU Union.
    }
}
